core/http_client.go

```diff
@@ -26,11 +26,23 @@ import (
 	"net/http"
 )
 
+// RemoteServerError describes an error returned when invoking a remote server.
+type RemoteServerError struct {
```

Comment:
I thought it was a custom error with specific contents. But it's actually a HttpError (>= 400)